python/cudf/cudf/core/buffer.py

```diff
@@ -72,23 +71,52 @@ def __init__(
                 raise TypeError("data must be Buffer, array-like or integer")
             self._init_from_array_like(np.asarray(data), owner)
 
+    @classmethod
+    def from_buffer(cls, buffer: Buffer, size: int = None, offset: int = 0):
+        """
+        Create a buffer from another buffer
+
+        Parameters
+        ----------
+        buffer : Buffer
+            The base buffer, which will also be set as the owner of
+            the memory allocation.
+        size : int, optional
+            Size of the memory allocation (default: `buffer.size`).
+        offset : int, optional
+            Start offset relative to `buffer.ptr`.
+        """
+
+        ret = cls()
+        ret._ptr = buffer._ptr + offset
+        ret._size = buffer.size if size is None else size
+        ret._owner = buffer
+        return ret
+
     def __len__(self) -> int:
-        return self.size
+        return self._size
+
+    @property
+    def size(self) -> int:
```

Comment:
I'd like to avoid having two public properties that are synonyms (`size` and `nbytes`). Is `size` needed as a public property in this PR? If not, can be it removed? If so, can we refactor and remove one of these two in the future?

Reply:
I guess it is to be compatible with [`rmm.DeviceBuffer`](https://github.com/rapidsai/rmm/blob/branch-22.06/python/rmm/_lib/device_buffer.pyx#L109). cuDF uses both `size` and `nbytes` but I'm happy to refactor cuDF to only use one of them, which one do we prefer?

Reply:
If we can remove the `size` property from this PR without causing tests to fail, that's what I would do for the moment.

Reply:
We cannot remove `size` nor `nbytes` without tests failing.